### PR TITLE
feat(teams): add members to team status

### LINKED ISSUE #723: [FEAT] - Refactor Team controller to store members in status
### Priority

None

### User Story

As a Member of an Organization I can see the members of Teams in the status of a Team so that I do not have to know there are two resources.

### Description

There are two CRDs that model a Team and it's Members (Team & TeamMembership).
The Team defines the upstream Group to be synced from SCIM. The TeamMembership contains the actual members of the Team. In terms of CRDs the Team is the Spec and the TeamMembership acts like the Status.

In order to simplify the controllers and to deprecate the TeamMembership controller in the future, the Team CRD should be extended so that it contains the Members in it's status. 
For a brief period the Members will be stored in both, the Team's status and in the TeamMemberships Spec. After the Admin UI has been migrated to retrieve the TeamMembers from the Status we can deprecate and remove the TeamMembership controller.

The `.Status` for `Team` should contain the list of members that is also part of the `.TeamMembership.Spec` 

### Acceptance Criteria

- [x] TeamMembership controller stays as is
- [x] Team Controller updates the Team's status with the members retrieved from SCIM

### Reference Issues

_No response_

### PR DESCRIPTION
**Pull request by gciezkowski-acc**
## Description
This PR add team members to a team status.

## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Added tests?

- [X] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [ ] 🙅 no documentation needed
- [X] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pkg/controllers/team/team_controller.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+func hasOwnerReference(ownerReferences []v1.OwnerReference, kind, name string) bool {
+	for _, ownerReference := range ownerReferences {
+		if ownerReference.Kind == kind && ownerReference.Name == name {
+			return true
+		}
+	}
+
+	return false
+}
```
Comment (abhijith-darshan) [inline pkg/controllers/team/team_controller.go:81]: can be replaced with 

https://github.com/cloudoperators/greenhouse/blob/2a5d694b37130578d372a43c7a2a86d1d892e367/pkg/clientutil/owner_reference.go#L10-L18

Comment (gciezkowski-acc) [reply pkg/controllers/team/team_controller.go]: Done

**Comment by abhijith-darshan** [comment]
There’s again the issue of two controllers reconciling the same resource. 

https://github.com/cloudoperators/greenhouse/blob/98a8368ce99db1d5433828cfe94cc758cc7d33ba/pkg/controllers/teammembership/teammembership_updater_controller.go#L77

**Review by abhijith-darshan — CHANGES_REQUESTED** [review]
there is again the issue of two controllers reconciling the same resource.

https://github.com/cloudoperators/greenhouse/blob/98a8368ce99db1d5433828cfe94cc758cc7d33ba/pkg/controllers/teammembership/teammembership_updater_controller.go#L77

File: charts/manager/templates/webhooks.yaml [hunk 2]
```diff
@@ -4,392 +4,380 @@ kind: MutatingWebhookConfiguration
...
+    - ""
+    apiVersions:
+    - v1
+    operations:
+    - CREATE
+    - UPDATE
+    - DELETE
+    resources:
+    - secrets
+  sideEffects: None
```
[the remark this answers is not in the document]
Comment (gciezkowski-acc) [reply charts/manager/templates/webhooks.yaml]: Done

**Comment by gciezkowski-acc** [comment]
> there is again the issue of two controllers reconciling the same resource.
> 
> https://github.com/cloudoperators/greenhouse/blob/98a8368ce99db1d5433828cfe94cc758cc7d33ba/pkg/controllers/teammembership/teammembership_updater_controller.go#L77

@abhijith-darshan I added members to team directly from SCIM Client.

**Review by IvoGoman — APPROVED** [review]

**Pull request merged**